File: lib-base/src/main/java/com/llj/lib/base/IActivityStack.java
package com.llj.lib.base;

import android.app.Activity;

import com.llj.lib.utils.AActivityStackUtils;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
public interface IActivityStack {

    default Activity getCurrentActivity() {
        return AActivityStackUtils.getCurrentActivity();
    }

    /**
     * 生成Activity存入列表
     *
     * @param activity
     */
    default void addCurrentActivity(Activity activity) {
        AActivityStackUtils.addCurrentActivity(activity);
    }

    /**
     * 移除当前的activity
     *
     * @param activity
     */
    default void removeCurrentActivity(Activity activity) {
        AActivityStackUtils.removeCurrentActivity(activity);
    }

    /**
     * 获得顶层下面的activity
     *
     * @return
     */
    default Activity getPreviousActivity() {
        return AActivityStackUtils.getPreviousActivity();
    }

    /**
     * 清除最上层以下所有的activity
     */
    default void clearBottomActivities() {
        AActivityStackUtils.clearBottomActivities();
    }


    /**
     * 清除所有的activity
     */
    default void removeAllActivity() {
        AActivityStackUtils.removeAllActivity();
    }

    default void exitApp() {
        AActivityStackUtils.exitApp();
    }

}

File: lib-base/src/main/java/com/llj/lib/base/IActivityStack.kt
package com.llj.lib.base

import android.app.Activity

import com.llj.lib.utils.AActivityStackUtils

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
interface IActivityStack {

    /**
     * 获得当前最顶层的activity
     *
     * @return 当前最顶层的activity
     */
    fun getCurrentActivity(activity: Activity): Activity? {
        return AActivityStackUtils.getCurrentActivity()
    }


    /**
     * 获得顶层下面的activity
     *
     * @return
     */
    fun getPreviousActivity(activity: Activity): Activity? {
        return AActivityStackUtils.getPreviousActivity()
    }


    /**
     * 生成Activity存入列表
     *
     * @param activity
     */
    fun addCurrentActivity(activity: Activity) {
        AActivityStackUtils.addCurrentActivity(activity)
    }

    /**
     * 移除当前的activity
     *
     * @param activity
     */
    fun removeCurrentActivity(activity: Activity) {
        AActivityStackUtils.removeCurrentActivity(activity)
    }

    /**
     * 清除最上层以下所有的activity
     */
    fun clearBottomActivities() {
        AActivityStackUtils.clearBottomActivities()
    }


    /**
     * 清除所有的activity
     */
    fun removeAllActivity() {
        AActivityStackUtils.removeAllActivity()
    }

    fun exitApp() {
        AActivityStackUtils.exitApp()
    }

}
